docs/Development-Guides/Rate-Limits.md

```diff
@@ -29,3 +22,166 @@ Product | Max requests initiated per minute
+
+Product | Max requests initiated per minute
+-------|-------------
+`company` | 20
+`directory` | 20
+`individual` | 20
+`employment` | 20
+`payment` | 12
+`pay-statement` | 12
+
+## IP Address Rate Limits
+
+Finch also enforces individual IP Address rate limits. An IP address is a unique address that identifies a device on the internet or a local network. If the number of `max requests` are sent from the same IP Address within the `duration` time set, a `penalty` is enforced. No more requests are allowed once the penalty is enforced. Once the penalty duration is complete, requests will be accepted again.
+
+Type | Max requests | Duration | Penalty
+-------|-------------|-------|-------------
+`API` | 1000 | 5 minutes | 60 minutes
+
+## Other types of rate limits
+
+### Finch Connect Rate limits
+
+Finch enforces rate limits on Employers going thru [Finch Connect](/docs/Integrating-with-Finch/Integrate-Finch-Connect/Overview.md). Each "step" in Finch Connect (the provider selection, logging in, multi-factor authentication prompts, etc) makes a single request to the Finch `/auth/authorize` endpoint. It is highly unlikely for an Employer to encounter these rate limit under normal use. However, an example of this rate limit being encountered is if an employer forgot their password and tried 30 different passwords in the time period of 1 minute. In this case, they would receive a rate limit penalty of 5 minutes.
+
+Type | Max requests | Duration | Penalty
+-------|-------------|-------|-------------
+`Auth` | 30 | 1 minute | 5 minutes
+
+### Upstream Provider Rate Limits
+
+Because Finch integrates directly with 180+ providers, we take extreme care to avoid provider-specific rate limits. However, it inevitably happens. In this case, we will return a specific error code [upstream_rate_limit_exceeded](/docs/Development-Guides/Errors/Error-Types.md#error-types-1) explaining that Finch encountered a provider-specific rate limit. In these cases, wait a few moments before retrying. It is difficult to recommend a wait duration since every provider is different.
+
+## Scenarios
+
+### Scenario 1: Hitting access token level rate limits
+
+In this scenario, let's focus on a single access token (Token A) and assume you are making API requests to any of the the `company`, `directory`, `individual`, `employment`, `payment`, and `pay-statement` endpoints. When a request  is sent to an endpoint, a single gallon of water is added to the endpoint’s bucket each time. The first requests starts that bucket’s 60-second Time-To-Live (TTL) timer.
+
+1. Token A makes `4` requests to the `directory` endpoint within a minute pouring 4 gallons of water into the “directory” bucket.
```

Comment:
Instead of "`4`" I'd write out 4 and reserve the code fence for actual code (same for the other numbers below.